packages/core/src/ui/UICanvas.ts

```
@@ -0,0 +1,498 @@
+    this._onCanvasSizeListener = this._onCanvasSizeListener.bind(this);
+    this._onCameraPropertyListener = this._onCameraPropertyListener.bind(this);
+    this._onCameraTransformListener = this._onCameraTransformListener.bind(this);
+    this._onReferenceResolutionChanged = this._onReferenceResolutionChanged.bind(this);
+    // @ts-ignore
+    this._referenceResolution._onValueChanged = this._onReferenceResolutionChanged;
+  }
+
+  _prepareRender(context: RenderContext): void {
+    const { renderers } = this;
+    const { frameCount } = this.engine.time;
+    const renderElement = (this._renderElement = this.engine._renderElementPool.get());
```

Biome

[error] The assignment should not be in an expression.

The use of assignments in expressions is confusing.
Expressions are often considered as side-effect free.

(lint/suspicious/noAssignInExpressions)